hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/FunctionalIndexSupport.scala

```diff
@@ -198,6 +209,44 @@ class FunctionalIndexSupport(spark: SparkSession,
 
     columnStatsRecords
   }
+
+  private def getPrunedPartitionsAndFileNamesMap(prunedPartitionsAndFileSlices: Seq[(Option[BaseHoodieTableFileIndex.PartitionPath], Seq[FileSlice])],
```

Comment:
can we make these package private/protected to get some UTs on them?

Reply:
Addressed